### PR TITLE
Checking if the result of a query or query_one function is not empty or nullptr

### PR DESCRIPTION
**Pull request by matebelenyesi**

### CODE REVIEW

File: plugins/cpp/service/src/cppservice.cpp [hunk 1]
```diff
@@ -1122,6 +1166,9 @@ void CppServiceHandler::getSyntaxHighlight(
     model::FilePtr file = _db->query_one<model::File>(
       FileQuery::id == std::stoull(range_.file));
 
+    if(!file)
+      return;
```
Comment (mcserep) [inline plugins/cpp/service/src/cppservice.cpp:1170]: Should be joined with the following check.

**Review by mcserep — CHANGES_REQUESTED** [review]
Looks good to me. Since these cases should not occur, maybe add some warning messages to the output, so we can easily identify them later?

File: plugins/cpp/service/src/cppservice.cpp [hunk 2]
```diff
@@ -892,7 +929,8 @@ void CppServiceHandler::getReferences(
           AstResult result = _db->query<model::CppAstNode>(
             AstQuery::entityHash == fr.theFriend &&
             AstQuery::astType == model::CppAstNode::AstType::Definition);
-          nodes.insert(nodes.end(), result.begin(), result.end());
+          if(!result.empty())
+            nodes.insert(nodes.end(), result.begin(), result.end());
```
[the remark this answers is not in the document]
Comment (zporky) [reply plugins/cpp/service/src/cppservice.cpp:933]: Yes, it is ok to v.insert( v.end(), ...) 
Moreover, if we are about to insert the first element to the vector than we have an empty vector. So this fix may deny us to insert one (or more) elements.

**Review by mcserep — APPROVED** [review]
Fixed the requested changes in bc22f919d564e1319a58044c87ff367ef00a148a.

**Pull request merged**